lib/intercom_flutter.dart

```diff
@@ -27,22 +27,22 @@ class Intercom {
   }
 
   static Future<dynamic> registerIdentifiedUser({String userId, String email}) {
-    if (userId.isNotEmpty) {
-      assert(email.isEmpty,
+    if (userId != null) {
+      assert(email == null,
           'The parameter `email` must be null if `userId` is provided.');
       return _channel.invokeMethod('registerIdentifiedUser', {
         'userId': userId,
       });
-    } else if (email.isNotEmpty) {
-      assert(userId.isEmpty,
+    } else if (email != null) {
+      assert(userId == null,
           'The parameter `userId` must be null if `email` is provided.');
       return _channel.invokeMethod('registerIdentifiedUser', {
         'email': email,
       });
     } else {
-      assert(userId.isNotEmpty || email.isNotEmpty,
+      assert(userId == null || email == null,
           'An identification method must be provided as a parameter, either `userId` or `email`.');
-      return registerUnidentifiedUser();
+      return _channel.invokeMapMethod('registerUnidentifiedUser');
```

Comment:
If no `userId` or `email` is provided, I added a call to `registerUnidentifiedUser`. In debug mode it will throw the `assert` but maybe in release mode, if no arguments are provided, I think we should register an Unidentified user (I don't know how Intercom handles this case).

Reply:
If the developer asks to do X in a wrong way, the package shouldnt do Y. Instead, throw an exception that X was called with invalid arguments.

So, I wouldnt call registerUnidientifiedUser when there is no argument given to registerIdentifiedUser. Instead return an exception.

Reply:
This makes sense. Might be a case of a null/empty variable passed in

Reply:
Solved, now it returns an `ArgumentError` if no or both parameters are passed in.